src/pudl/transform/ferc1.py

```diff
@@ -2869,66 +2869,25 @@ def process_dbf(self, raw_dbf):
-            table_name: name of table
-            source_ferc1: data source of raw ferc1 database.
+    def source_table_id(
+        self, source_ferc1: SourceFerc1, df: pd.DataFrame | None = None
+    ) -> str | pd.Series:
+        """Look up the ID of the raw data source table.
 
-        Raises:
-            ValueError: If any of the primary key columns are missing from the DataFrame
-                being processed.
-            ValueError: If there are any null values in the primary key columns.
-            ValueError: If the resulting `record_id` column is non-unique.
+        Because the raw DBF data comes from two seperate tables, this table-specific
+        method generates a Series of tables names based on the ``sched_table_name``
+        columns which was assigned during the transform step. For the XBRL source, the
+        standard method is employed and a string is returned.
         """
-        logger.debug(
-            f"{self.table_id.value}: Assigning {source_ferc1.value} source record IDs."
-        )
-        pk_cols = self.renamed_table_primary_key(source_ferc1)
-        missing_pk_cols = set(pk_cols).difference(df.columns)
-        if missing_pk_cols:
-            raise ValueError(
-                f"{self.table_id.value} ({source_ferc1.value}): Missing primary key "
-                "columns in dataframe while assigning source record_id: "
-                f"{missing_pk_cols}"
-            )
-        if df[pk_cols].isnull().any(axis=None):
-            raise ValueError(
-                f"{self.table_id.value} ({source_ferc1.value}): Found null primary key "
-                "values.\n"
-                f"{df[pk_cols].isnull().any()}"
-            )
         # assign the source_table_id column based on the source
         if source_ferc1 == SourceFerc1.DBF:
             # sched_table_name is a table name. was added in align_row_numbers_dbf
-            df.sched_table_name.isnull().any()
-            df = df.assign(source_table_id=lambda x: x.sched_table_name)
+            if df.sched_table_name.isnull().any():
+                raise AssertionError(f"BAd. {df.sched_table_name.isnull().any()}")
+            return df.sched_table_name
         else:
             assert source_ferc1 == SourceFerc1.XBRL
-            df = df.assign(source_table_id=self.source_table_id(source_ferc1))
-        df = df.assign(
-            record_id=lambda x: x.source_table_id.str.cat(
-                x[pk_cols].astype(str), sep="_"
-            ),
-        ).drop(columns=["source_table_id"])
-        df.record_id = enforce_snake_case(df.record_id)
-
-        dupe_ids = df.record_id[df.record_id.duplicated()].values
-        if dupe_ids.any() and self.has_unique_record_ids:
-            logger.warning(
-                f"{self.table_id.value}: Found {len(dupe_ids)} duplicate record_ids: \n"
-                f"{dupe_ids}."
-            )
-        return df
+            return super().source_table_id(source_ferc1=source_ferc1)
```

Comment:
I think it's a little clearer when there are multiple branches like this to assign to the variable that's going to be returned, and then have a single return statement at the end which is outside of any conditional. It also guarantees that you actually get a returned value, in the event that there are unexpected conditions that result in none of the returns inside the conditionals getting run.

Reply:
yeah that makes sense! as you can maybe tell i did a first pass on this method to make sure ti worked and didn't reeeeeally go back and tidy it. this has been edited